.github/workflows/build.yml

```
@@ -2,12 +2,22 @@ name: Python Fire
 
 on: [push]
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.5", "3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
+        fail-fast: false
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
```

Starting slower ones first can prevent bottlenecks and help builds as a whole finish sooner.

```suggestion
        os: ["windows-latest", "macos-latest", "ubuntu-latest"]
```